spi/src/main/java/org/eclipse/dataspaceconnector/spi/Result.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright (c) 2021 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.spi;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+
+public class Result<T> {
+
+    public static <T> Result<T> success(@NotNull T content) {
+        return new Result<>(content, emptyList());
+    }
+
+    public static <T> Result<T> failure(String failure) {
+        return new Result<>(null, List.of(failure));
+    }
+
+    public static <T> Result<T> failure(List<String> failures) {
```

Comment:
A nit: this should probably be "failures(...)"

Reply:
I thought it more in the way of: "This result represent a failure, and here's the detail" in opposition of "this result represent a success"

Reply:
Maybe it should be "problem" or "error detail" then?